spec/features/discounts/new_spec.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'As a merchant employee', type: :feature do
+  describe 'when I visit my merchant dashboard' do
+    before(:each) do
+      @merchant_user = User.create!(name: 'Alex',
+                                    street_address: '321 Jones Dr',
+                                    city: 'Jonesville',
+                                    state: 'WI',
+                                    zip_code: 54_321,
+                                    email: 'alex@example.com',
+                                    password: 'alex123',
+                                    role: 1)
+
+      @merchant1 = create(:random_merchant)
+      @merchant1.users << @merchant_user
+
+      @merchant2 = create(:random_merchant)
+
+      @discount1 = @merchant1.discounts.create!(name: 'winter special', number_items: 10, percent: 15)
+      @discount2 = @merchant1.discounts.create!(name: 'big sale', number_items: 5, percent: 10)
+      @discount3 = @merchant2.discounts.create!(name: 'Huge sale', number_items: 30, percent: 20)
+
+      visit '/login'
+      fill_in :email, with: @merchant_user.email
+      fill_in :password, with: @merchant_user.password
+      click_button 'Log In'
+    end
+
+    it 'there is a link to create a new discount for my merchant' do
+      visit '/merchant/discounts'
+
+      click_link('Add Discount')
+      expect(current_path).to eq('/merchant/discounts/new')
+
+      fill_in :name, with: 'Super Sale'
+      fill_in :number_items, with: 15
+      fill_in :percent, with: 17
+
+      last_discount = @merchant1.discounts.last
+
+      click_button 'Submit'
+
+      new_discount = @merchant1.discounts.last
```

Comment:
checking that the last discount created changes after clicking the `submit` button when the discount is valid